src/artemis/devices/zebra.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import annotations
+from typing import Iterator, List, Dict
+from enum import Enum
+
+from ophyd import Component, Device, EpicsSignal, StatusBase
+from ophyd.status import SubscriptionStatus
+
+from functools import partial, partialmethod
+
+PC_ARM_SOURCE_SOFT = 0
+PC_ARM_SOURCE_EXT = 1
+
+PC_GATE_SOURCE_POSITION = 0
+PC_GATE_SOURCE_TIME = 1
+PC_GATE_SOURCE_EXTERNAL = 2
+
+PC_PULSE_SOURCE_POSITION = 0
+PC_PULSE_SOURCE_TIME = 1
+PC_PULSE_SOURCE_EXTERNAL = 2
+
+# Sources
+DISCONNECT = 0
+IN1_TTL = 1
+IN2_TTL = 4
+PC_ARM = 29
+PC_GATE = 30
+PC_PULSE = 31
+AND3 = 34
+AND4 = 35
+OR1 = 36
+PULSE1 = 52
+SOFT_IN3 = 62
+
+# Instrument specific
+TTL_DETECTOR = 2
+TTL_XSPRESS3 = 3
+TTL_SHUTTER = 4
+
+
+def epics_signal_put_wait(pv_name: str, wait: str = 1.0) -> EpicsSignal:
+    """Creates a `Component` around an `EpicsSignal` that waits for a callback on a put.
+
+    Args:
+        pv_name (str): The name of the PV for the `EpicsSignal`
+        wait (str, optional): The timeout to wait for a callback. Defaults to 1.0.
+
+    Returns:
+        EpicsSignal: An EpicsSignal that will wait for a callback.
+    """
+    return Component(EpicsSignal, pv_name, put_complete=True, write_timeout=wait)
+
+
+class PositionCompare(Device):
+    num_gates: EpicsSignal = epics_signal_put_wait("PC_GATE_NGATE")
+    gate_source: EpicsSignal = epics_signal_put_wait("PC_GATE_SEL")
+    gate_input: EpicsSignal = epics_signal_put_wait("PC_GATE_INP")
+
+    pulse_source: EpicsSignal = epics_signal_put_wait("PC_PULSE_SEL")
+    pulse_input: EpicsSignal = epics_signal_put_wait("PC_PULSE_INP")
+
+    dir: EpicsSignal = Component(EpicsSignal, "PC_DIR")
+    arm_source: EpicsSignal = epics_signal_put_wait("PC_ARM_SEL")
+    arm_demand: EpicsSignal = Component(EpicsSignal, "PC_ARM")
+    disarm_demand: EpicsSignal = Component(EpicsSignal, "PC_DISARM")
+    armed: EpicsSignal = Component(EpicsSignal, "PC_ARM_OUT")
+
+    def setup_fast_grid_scan(self):
+        self.arm_source.put(PC_ARM_SOURCE_SOFT)
+        self.gate_source.put(PC_GATE_SOURCE_EXTERNAL)
+
+        # Set up parameters for the GATE
+        self.gate_input.put(SOFT_IN3)
+        self.num_gates.put(1)
+
+        # Pulses come in through TTL input 1
+        self.pulse_source.put(PC_PULSE_SOURCE_EXTERNAL)
+        self.pulse_input.put(IN1_TTL)
+
+    def arm(self) -> StatusBase:
+        status = self.arm_status(1)
+        self.arm_demand.put(1)
```

Comment:
Within GDA a wait is added until the Zebra is armed.  (Noted that callback not wanted from the arm demand as not all Zebra IOCs send a callback)

Reply:
I see now waits added in Zebra stage and unstage methods